src/kontrol/foundry.py

```diff
@@ -305,20 +305,45 @@ def matching_sigs(self, test: str) -> list[str]:
         return test_sigs
 
     def get_test_id(self, test: str, version: int | None) -> str:
+        """
+        Retrieves the unique identifier for a test based on its name and version.
+
+        If multiple proofs are found for a test without a specific version, the function attempts to resolve to the latest version.
+
+        :param test: The name of the test to find a matching proof for.
+        :param version: The version number of the test. If None, the function attempts to resolve to the latest version if multiple matches are found.
+        :raises ValueError: If no matching proofs are found for the given test and version, indicating the test does not exist.
+        :raises ValueError: If more than one matching proof is found for a given test and version, a full signature is required.
+        :return: The unique identifier of the matching proof for the specified test and version.
+        """
+
+        def _assert_single_id(l: list[str]) -> str:
+            try:
+                return single(l)
+            except ValueError as e:
+                error_msg = (
+                    f'Found {len(matching_proof_ids)} matching proofs for {test}:{version}. '
+                    f'Provide a full signature of the test, e.g., {matching_sigs[0][5:]!r} --version {version}. '
+                    f'Error: {e}'
+                )
+                raise ValueError(error_msg) from e
+
         matching_proof_ids = self.proof_ids_with_test(test, version)
-        sig = single(self.matching_sigs(test))
-        if len(matching_proof_ids) == 0:
+        matching_sigs = self.matching_sigs(test)
+
+        if not matching_proof_ids:
             raise ValueError(f'Found no matching proofs for {test}:{version}.')
-        if len(matching_proof_ids) > 1:
-            if version is None:
-                raise ValueError(
-                    f'Found {len(matching_proof_ids)} matching proofs for {test}:{version}. Use the --version flag to choose one.'
-                )
-            else:
-                raise ValueError(
-                    f'Found {len(matching_proof_ids)} matching proofs for {test}:{version}. Provide a full signature of the test, e.g., {sig[5:]!r} --version {version}.'
-                )
```

Comment:
This branch was never reached because `single(matching_sigs)` would throw first.